myriad-scheduler/src/main/java/org/apache/myriad/Main.java

```diff
@@ -156,13 +157,19 @@ private void initProfiles(Injector injector) {
     ServiceProfileManager profileManager = injector.getInstance(ServiceProfileManager.class);
     Map<String, Map<String, String>> profiles = injector.getInstance(MyriadConfiguration.class).getProfiles();
     TaskUtils taskUtils = injector.getInstance(TaskUtils.class);
+    OfferUtils.setVcoreRatio(taskUtils.getNodeManagerVcoreRatio());
     if (MapUtils.isNotEmpty(profiles)) {
       for (Map.Entry<String, Map<String, String>> profile : profiles.entrySet()) {
         Map<String, String> profileResourceMap = profile.getValue();
-        if (MapUtils.isNotEmpty(profiles) && profileResourceMap.containsKey("cpu") && profileResourceMap.containsKey("mem")) {
-          Long cpu = Long.parseLong(profileResourceMap.get("cpu"));
+        if (MapUtils.isNotEmpty(profiles) &&
+            (profileResourceMap.containsKey("vcore") || profileResourceMap.containsKey("cpu")) &&
+            profileResourceMap.containsKey("mem")) {
+          Long vcore = profileResourceMap.containsKey("vcore") ?
+              Long.parseLong(profileResourceMap.get("vcore")) :
+              Long.parseLong(profileResourceMap.get("cpu"));
```

Comment:
@yufeldman 
I understand your concern. In my opinion, when we separate cpu and vcore params, it would be more confusing. It seems to me that we should set exactly two of the three params (cpu, vcore, ratio) but could not set all of them at the same time. 
Also it seems that we could set both cpu and vcore at the same time, but ratio is not set. And the ratio of small NM and medium NM may not be the same. As a result, the real cpu of one vcore on NMs would be different.